src/NHibernate/Linq/Visitors/HqlGeneratorExpressionVisitor.cs

```diff
@@ -540,7 +540,7 @@ protected HqlTreeNode VisitConditionalExpression(ConditionalExpression expressio
 
 			return (expression.Type == typeof (bool) || expression.Type == (typeof (bool?)))
 					   ? @case
-					   : _hqlTreeBuilder.Cast(@case, expression.Type);
+					   : _hqlTreeBuilder.TransparentCast(@case, expression.Type);
```

Comment:
Still a crutch, but at least one that causes less bugs.

I wonder if some other cast introduced by other method in this file should be switched to transparent too or not. Maybe the `Avg` by example, for it output cast (input casts should clearly stay actual casts). Or the `Sum` cast.